ldi-orchestrator/ldio-connectors/ldio-ldes-client/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/LdioLdesClient.java

```diff
@@ -109,17 +117,15 @@ protected void pause() {
 	}
 
 	private void shutdownPipeline() {
-		Thread.ofVirtual().name(LDIO_SHUTDOWN_THREAD_NAME).start(() -> {
-			while (Boolean.FALSE.equals(canGracefullyShutdownChecker.get())) {
-				try {
-					Thread.sleep(Duration.ofSeconds(30));
-				} catch (InterruptedException e) {
-					Thread.currentThread().interrupt();
-				}
+		threadRunning = false;
+		do {
+			try {
+				Thread.sleep(Duration.ofSeconds(1));
```

Comment:
Isn't that very intensive to run this every second?

Reply:
Keep in mind, this only happens when shutting down the client or when an exception is reached. 
In the case of a manual shutdown, you want to have your http response as soon as possible. 
Also, the throughput should be quite fast